typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -110,24 +153,76 @@ export class LiquidityLayerApp<
         txHash,
         message,
         nonce,
-        domain: 0,
+        domain,
         nonceHash: ethers.utils.solidityKeccak256(
           ['uint32', 'uint256'],
-          [0, nonce],
+          [domain, nonce],
         ),
       },
     ];
   }
 
+  async attemptPortalTransferCompletion(
+    message: PortalBridgeMessage<Chain>,
+  ): Promise<void> {
+    const destinationPortalAdapter = this.getContracts(message.destination)
+      .portalAdapter!;
+
+    const transferId = await destinationPortalAdapter.transferId(
+      ChainNameToDomainId[message.origin],
+      message.nonce,
+    );
+
+    const transferTokenAddress =
+      await destinationPortalAdapter.portalTransfersProcessed(transferId);
+
+    if (!utils.eqAddress(transferTokenAddress, ethers.constants.AddressZero)) {
+      console.log(
+        `Transfer with nonce ${message.nonce} from ${message.origin} to ${message.destination} already processed`,
+      );
+      return;
+    }
+
+    const wormholeOriginDomain = this.config[
+      message.destination
+    ].portal!.wormholeDomainMapping.find(
+      (_) => _.hyperlaneDomain === ChainNameToDomainId[message.origin],
+    )?.wormholeDomain;
+    const emitter = utils.strip0x(
+      utils.addressToBytes32(
+        this.config[message.origin].portal!.portalBridgeAddress,
+      ),
+    );
+
+    const vaa = await fetch(
+      `${PORTAL_VAA_SERVICE_TESTNET_BASE_URL}${wormholeOriginDomain}/${emitter}/${message.portalSequence}`,
+    ).then((_) => _.json());
+
+    if (vaa.code && vaa.code === 5) {
```

Comment:
why 5?

Reply:
Magic constant on their side, but I'll pull out in a constant